roboteam_world/src/Handler.cpp

```diff
@@ -31,6 +33,23 @@ void Handler::start(std::string visionip, std::string refereeip, int visionport,
         std::string file_name = "world_log_" + std::to_string(days) + "_" +std::to_string(hours) + "_" + std::to_string(minutes) +"_" + std::to_string(seconds) + "_" + std::to_string(mili)+".log";
         fileWriter = rtt::LogFileWriter();
         fileWriter.value().open(file_name);
+        //Create a signal handler so that when we crash data is still saved,
+        // so that the file is still closed when the program is unexpectedly terminated or an error is thrown
+        //If we do not do this, the filestream buffer may contain the messages which caused the crash,
+        // making it impossible to debug the issue afterwards
+        auto signalHandler = [](int signum){
+            if(Handler::fileWriter.has_value()){
+                Handler::fileWriter->close();
```

Comment:
Should this be in a try-catch block? What happens if an exception is thrown in a signal handler?

Reply:
Afaik the signal handler cannot throw exceptions, but I'll do so to make sure that things cannot go south.

Reply:
I looked into it (briefly) and it seems you're right.